miniwin/miniwin/src/miniwin_ddraw.cpp

```diff
@@ -0,0 +1,181 @@
+#include "miniwin_ddraw.h"
+
+#include "miniwin_d3d.h"
+
+#include <SDL3/SDL.h>
+
+struct DirectDrawImpl : public IDirectDraw2, public IDirect3D2 {
+	DirectDrawImpl() : m_refCount(1) {}
+	// IUnknown interface
+	ULONG AddRef() override
+	{
+		m_refCount += 1;
+		return m_refCount;
+	}
+	ULONG Release() override
+	{
+		m_refCount -= 1;
+		if (m_refCount == 0) {
+			delete this;
```

Comment:
Fixes use after free:
```suggestion
			delete this;
			return 0;
```

Reply:
This is indeed a use-after-free :) Good thing we have that.